crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -32,6 +32,55 @@ def role_constraints
     end
   end
 
+  def shutdown_openstack_at_nodes
+    db_nodes = []
+    NodeObject.all.each do |node|
+      next unless node.state == "crowbar_upgrade"
+
+      # Find the nodes with Database role
+      # (we can't look at node.roles, because that list has been reduced to just crowbar-upgrade)
+      node.crowbar["run_list_map"].each do |role, v|
+        db_nodes.push node.name if role == "database-server"
+      end
+
+      # mark the position in the upgrade process
+      node["crowbar_wall"]["crowbar_openstack_shutdown"] = true
+      node.save
+    end
+
+    # Adapt current proposal, so only non-db nodes have crowbar-upgrade role
+    # (crowbar-upgrade role is currently assigned to all nodes)
+    proposal = ProposalObject.find_proposal("crowbar", "default")
+    proposal["deployment"]["crowbar"]["elements"]["crowbar-upgrade"] -= db_nodes
```

Comment:
Or you mean that I should run the first round on all nodes, not just DB ones? That's what I also considered, but in that case I would need to add additional checks if node has db role or not into the recipe and I thought that assigning DB nodes different role directly makes it cleaner.